webapp/components/new-test-results-history-grid.js

```diff
@@ -0,0 +1,220 @@
+    this.dataTable.addColumn({ type: 'string', id: 'Status' });
+
+    // style and tooltip columns that are not displayed
+    this.dataTable.addColumn({ type: 'string', id: 'style', role: 'style' });
+    this.dataTable.addColumn({ type: 'string', role: 'tooltip' });
+
+    this.dataTable.addColumn({ type: 'date', id: 'Start' });
+    this.dataTable.addColumn({ type: 'date', id: 'End' });
+
+    const dataTableRows = [];
+    const now = new Date();
+    this.runIDs = [];
+
+    // Create a row for each subtest
+    this.subtestNames.forEach(subtestName => {
+      for (let i = 0; i < browserTestData[subtestName].length; i++) {
+        const dataPoint = browserTestData[subtestName][i];
+        const startDate = new Date(Number(dataPoint.date));
+
+        // Use the next entry as the end date, or use present time if this
+        // is the last entry
+        let endDate = now;
+        if (i + 1 !== browserTestData[subtestName].length) {
+          const nextDataPoint = browserTestData[subtestName][i + 1];
+          endDate = new Date(Number(nextDataPoint.date));
+        }
+
+        // If this is the main test status, name it based on the amount of subtests
+        let subtestDisplayName = subtestName;
+        if (subtestName === '') {
+          subtestDisplayName = (this.subtestNames.length > 1) ? 'Harness status' : 'Test status';
+        }
+
+        const tooltip =
+          `${dataPoint.status} ${startDate.toLocaleDateString()}-${endDate.toLocaleDateString()}`;
+        const statusColor = COLOR_MAPPING[dataPoint.status] || COLOR_MAPPING.default;
+
+        // Add the run ID to array of run IDs to use for links
+        this.runIDs.push(dataPoint.run_id);
+
+        dataTableRows.push([
+          subtestDisplayName,
+          dataPoint.status,
+          statusColor,
+          tooltip,
+          startDate,
+          endDate,
+        ]);
+      }
+    });
+
+    let options = {
+      tooltip: {
+        isHtml: false,
+      },
+    };
+    this.dataTable.addRows(dataTableRows);
+
+    // handler to allow rows to be clicked and navigate to the run url
+    // https://stackoverflow.com/questions/40928971/how-to-customize-google-chart-with-hyperlink-in-the-label
+    const statusSelectHandler = () => {
+      const selection = this.chart.getSelection();
+      if (selection.length > 0) {
+        const index = selection[0].row;
+        if (index !== undefined && this.runIDs.length > index) {
+          window.open(`/results/?run_id=${this.runIDs[index]}`, '_blank');
+        }
+      }
+    };
+    window.google.visualization.events.addListener(this.chart, 'select', statusSelectHandler);
+
+    this.chart.draw(this.dataTable, options);
+  }
+
+  // get test history and aligned run data
+  async getTestHistory() {
+    this.historicalData = await fetch('/api/history')
+      .then(r => r.json()).then(data => data.results);
+    this.subtestNames = Object.keys(this.historicalData[0]);
```

Comment:
This might need to be changed after we're pulling true data, because it might not be the case that the data we're getting is sorted as we want it. However, this requires no action now.